codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSSigningParams.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.aws.swift.codegen
+
+import software.amazon.smithy.aws.traits.auth.UnsignedPayloadTrait
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.model.hasTrait
+import java.util.Locale
+
+data class AWSSigningParams(
+    // The service that the signature is being prepared for.
+    val service: ServiceShape,
+    // The operation that the signature is being prepared for.
+    val operation: OperationShape,
+    // When set to true, forces the AWSv4 signature to be placed in the query string.
+    val useSignatureTypeQueryString: Boolean,
+    // When set to true, the AWSv4 signature does not include the body of the request.
+    val forceUnsignedBody: Boolean,
+    // When set to true, the expiration is included in the params string so it may be set to a custom
+    // value named `expiration`.
+    val useExpiration: Boolean
+)
+
+class SigV4Configurator(
+    val useSignatureTypeQueryString: Boolean,
+    val useExpiration: Boolean,
+    val serviceIsS3: Boolean,
```

Comment:
Do we have to expose `serviceIsS3` here? Can we add values for `useDoubleURIEncode` and `shouldNormalizeURIPath` instead (and set those values accordingly in the constructor)?

This future proofs this a bit as it makes it easier to add customizations for other services in the future.

Reply:
Changed.  `SigV4Configurator` now knows nothing about services (except in the convenience initializer).  I added a few more tests to cover as well.